### PR TITLE
README improvements

### PR DESCRIPTION
**Pull request by AdrianSoundy**
## Description

Initial I thought I would just add a section for the ESP32 based ethernet boards as we build a number of them, but they are hidden. 

Then thought it might be cleaner if a separated the different ESP32 versions in to separate tables and added a note column.

On top I have added 2 versions of a jump list to the different sections of the list as it's getting very long. Probably need to merge to make sure jumps are working. 

Please review and let me know what you prefer.  
- No jump list
- Jump list as bullet points
- Jump list in table

Does it look better with ESP32 in sections?

Are the notes relevant?

I haven't made any changes to Chinese version, need review first, also I haven't updated class library feature table for new targets. This is sort of hidden as you have to expand. Maybe this should be a link to a documentation page or separate doc pages for each target.

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [ ] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dev Containers (changes related with Dev Containers, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [x] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist

- [x] My code follows the code style of this project (only if there are changes in source code).
- [ ] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [x] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [ ] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).

## Summary by CodeRabbit

- **Documentation**
  - Reorganized and reformatted documentation for various boards including ESP32, ESP32_S2, ESP32_S3, ESP32 RISC-V, M5Stack, STM32, Silicon Labs Giant Gecko, NXP, and TI boards.
  - Added tables for better readability and updated the presentation of target lists.

- **New Features**
  - Added version information for various ESP32 boards, including those with inbuilt Ethernet, ESP32_S2, ESP32_S3, ESP32 RISC-V, and M5Stack boards.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,43 +23,85 @@ Firmware versions contained below are released via the `main` branch, and are tr
 
 We also have a [Community Targets](https://github.com/nanoframework/nf-Community-Targets) repository where you can find firmware versions for several other popular boards that have been contibuted by the community, although limited support is provided.
 
+## Available targets
```
Comment (Ellerbach) [inline README.md:26]:
```suggestion
## Available targets

```

File: README.md [hunk 2]
```diff
@@ -23,43 +23,85 @@ Firmware versions contained below are released via the `main` branch, and are tr
...
+| ESP32_PSRAM_BLE_GenericGraphic_REV3 | |[![Latest Version @ Cloudsmith](https://api-prd.cloudsmith.io/v1/badges/version/net-nanoframework/nanoframework-images/raw/ESP32_PSRAM_BLE_GenericGraphic_REV3/latest/x/?render=true)](https://cloudsmith.io/~net-nanoframework/repos/nanoframework-images/packages/detail/raw/ESP32_PSRAM_BLE_GenericGraphic_REV3/latest/) |
+| ESP_WROVER_KIT | | [![Latest Version @ Cloudsmith](https://api-prd.cloudsmith.io/v1/badges/version/net-nanoframework/nanoframework-images/raw/ESP_WROVER_KIT/latest/x/?render=true)](https://cloudsmith.io/~net-nanoframework/repos/nanoframework-images/packages/detail/raw/ESP_WROVER_KIT/latest/) |
+| ESP32_PICO | | [![Latest Version @ Cloudsmith](https://api-prd.cloudsmith.io/v1/badges/version/net-nanoframework/nanoframework-images/raw/ESP32_PICO/latest/x/?render=true)](https://cloudsmith.io/~net-nanoframework/repos/nanoframework-images/packages/detail/raw/ESP32_PICO/latest/) |
+
+### ESP32 with inbuilt Ethernet 
+| Target | Note | Version | 
+|:-|---|---|
```
Comment (Ellerbach) [inline README.md:67]:
```suggestion
|:---|---|---|
```

File: README.md [hunk 3]
```diff
@@ -23,43 +23,85 @@ Firmware versions contained below are released via the `main` branch, and are tr
...
+| ESP32_WT32_ETH01  | | [![Latest Version @ Cloudsmith](https://api-prd.cloudsmith.io/v1/badges/version/net-nanoframework/nanoframework-images/raw/ESP32_WT32_ETH01/latest/x/?render=true)](https://cloudsmith.io/~net-nanoframework/repos/nanoframework-images/packages/detail/raw/ESP32_WT32_ETH01/latest/) |
+| ESP32_WESP32  | Poe | [![Latest Version @ Cloudsmith](https://api-prd.cloudsmith.io/v1/badges/version/net-nanoframework/nanoframework-images/raw/ESP32_WESP32/latest/x/?render=true)](https://cloudsmith.io/~net-nanoframework/repos/nanoframework-images/packages/detail/raw/ESP32_WESP32/latest/) |
+| ESP32_LILYGO | Poe | [![Latest Version @ Cloudsmith](https://api-prd.cloudsmith.io/v1/badges/version/net-nanoframework/nanoframework-images/raw/ESP32_LILYGO/latest/x/?render=true)](https://cloudsmith.io/~net-nanoframework/repos/nanoframework-images/packages/detail/raw/ESP32_LILYGO/latest/) |
+
+### ESP32_S2 boards
```
Comment (Ellerbach) [inline README.md:74]:
```suggestion
### ESP32_S2 boards

```

File: README.md [hunk 4]
```diff
@@ -23,43 +23,85 @@ Firmware versions contained below are released via the `main` branch, and are tr
...
+| Target | Note | Version | 
+|:-|---|---|
```
Comment (Ellerbach) [inline README.md:76]:
```suggestion
|:---|---|---|
```

File: README.md [hunk 5]
```diff
@@ -23,43 +23,85 @@ Firmware versions contained below are released via the `main` branch, and are tr
...
+| FEATHER_S2 | | [![Latest Version @ Cloudsmith](https://api-prd.cloudsmith.io/v1/badges/version/net-nanoframework/nanoframework-images/raw/FEATHER_S2/latest/x/?render=true)](https://cloudsmith.io/~net-nanoframework/repos/nanoframework-images/packages/detail/raw/FEATHER_S2/latest/) |
+| KALUGA_1 | Espressif | [![Latest Version @ Cloudsmith](https://api-prd.cloudsmith.io/v1/badges/version/net-nanoframework/nanoframework-images/raw/KALUGA_1/latest/x/?render=true)](https://cloudsmith.io/~net-nanoframework/repos/nanoframework-images/packages/detail/raw/KALUGA_1/latest/) |
+
+### ESP32_S3 boards
```
Comment (Ellerbach) [inline README.md:80]:
```suggestion
### ESP32_S3 boards

```

File: README.md [hunk 6]
```diff
@@ -23,43 +23,85 @@ Firmware versions contained below are released via the `main` branch, and are tr
...
+| Target | Note | Version | 
+|:-|---|---|
```
Comment (Ellerbach) [inline README.md:82]:
```suggestion
|:---|---|---|
```

File: README.md [hunk 7]
```diff
@@ -23,43 +23,85 @@ Firmware versions contained below are released via the `main` branch, and are tr
...
+| ESP32_S3 | Quad spiram support | [![Latest Version @ Cloudsmith](https://api-prd.cloudsmith.io/v1/badges/version/net-nanoframework/nanoframework-images/raw/ESP32_S3/latest/x/?render=true)](https://cloudsmith.io/~net-nanoframework/repos/nanoframework-images/packages/detail/raw/ESP32_S3/latest/) |
+| ESP32_S3_BLE | Display, BLE, Quad spiram support | [![Latest Version @ Cloudsmith](https://api-prd.cloudsmith.io/v1/badges/version/net-nanoframework/nanoframework-images/raw/ESP32_S3_BLE/latest/x/?render=true)](https://cloudsmith.io/~net-nanoframework/repos/nanoframework-images/packages/detail/raw/ESP32_S3_BLE/latest/) |
+| ESP32_S3_ALL |  Display, BLE, Octal spiram support | [![Latest Version @ Cloudsmith](https://api-prd.cloudsmith.io/v1/badges/version/net-nanoframework/nanoframework-images/raw/ESP32_S3_ALL/latest/x/?render=true)](https://cloudsmith.io/~net-nanoframework/repos/nanoframework-images/packages/detail/raw/ESP32_S3_ALL/latest/) |
+
+### ESP32 risc-v boards
+| Target |  Note | Version |
+|:-|---|---|
```
Comment (Ellerbach) [inline README.md:89]:
```suggestion
|:---|---|---|
```

File: README.md [hunk 8]
```diff
@@ -23,43 +23,85 @@ Firmware versions contained below are released via the `main` branch, and are tr
...
-| ESP32_S3 | [![Latest Version @ Cloudsmith](https://api-prd.cloudsmith.io/v1/badges/version/net-nanoframework/nanoframework-images/raw/ESP32_S3/latest/x/?render=true)](https://cloudsmith.io/~net-nanoframework/repos/nanoframework-images/packages/detail/raw/ESP32_S3/latest/) |
-| ESP32_S3_BLE | [![Latest Version @ Cloudsmith](https://api-prd.cloudsmith.io/v1/badges/version/net-nanoframework/nanoframework-images/raw/ESP32_S3_BLE/latest/x/?render=true)](https://cloudsmith.io/~net-nanoframework/repos/nanoframework-images/packages/detail/raw/ESP32_S3_BLE/latest/) |
-| ESP32_S3_ALL | [![Latest Version @ Cloudsmith](https://api-prd.cloudsmith.io/v1/badges/version/net-nanoframework/nanoframework-images/raw/ESP32_S3_ALL/latest/x/?render=true)](https://cloudsmith.io/~net-nanoframework/repos/nanoframework-images/packages/detail/raw/ESP32_S3_ALL/latest/) |
+| Target | Note | Version |
+|:-|---|---|
```
Comment (Ellerbach) [inline README.md:52]:
```suggestion
|:---|---|---|
```

File: README.md [hunk 9]
```diff
@@ -23,43 +23,85 @@ Firmware versions contained below are released via the `main` branch, and are tr
...
-| [M5StickCPlus](https://docs.m5stack.com/en/core/m5stickc_plus) | [![Latest Version @ Cloudsmith](https://api-prd.cloudsmith.io/v1/badges/version/net-nanoframework/nanoframework-images/raw/M5StickCPlus/latest/x/?render=true)](https://cloudsmith.io/~net-nanoframework/repos/nanoframework-images/packages/detail/raw/M5StickCPlus/latest/) |
-| [M5Core2](https://docs.m5stack.com/en/core/core2) | [![Latest Version @ Cloudsmith](https://api-prd.cloudsmith.io/v1/badges/version/net-nanoframework/nanoframework-images/raw/M5Core2/latest/x/?render=true)](https://cloudsmith.io/~net-nanoframework/repos/nanoframework-images/packages/detail/raw/M5Core2/latest/) |
-| [AtomS3](https://docs.m5stack.com/en/core/AtomS3) | [![Latest Version @ Cloudsmith](https://api-prd.cloudsmith.io/v1/badges/version/net-nanoframework/nanoframework-images/raw/AtomS3/latest/x/?render=true)](https://cloudsmith.io/~net-nanoframework/repos/nanoframework-images/packages/detail/raw/AtomS3/latest/) |
+| Target | Note | Version |
+|:-|---|---|
```
Comment (Ellerbach) [inline README.md:99]:
```suggestion
|:---|---|---|
```

**Review by Ellerbach — APPROVED** [review]
looks good, couple of linting propositions

File: README.md (hunk 1 above)
Comment (AdrianSoundy) [reply README.md:26]: Adding empty lines doesn't do anything to rendered output for markdown. The rendered output already has a space between header and table.

File: README.md [hunk 10]
```diff
@@ -23,43 +23,85 @@ Firmware versions contained below are released via the `main` branch, and are tr
...
+| ESP32_PSRAM_BLE_GenericGraphic_REV3 | |[![Latest Version @ Cloudsmith](https://api-prd.cloudsmith.io/v1/badges/version/net-nanoframework/nanoframework-images/raw/ESP32_PSRAM_BLE_GenericGraphic_REV3/latest/x/?render=true)](https://cloudsmith.io/~net-nanoframework/repos/nanoframework-images/packages/detail/raw/ESP32_PSRAM_BLE_GenericGraphic_REV3/latest/) |
+| ESP_WROVER_KIT | | [![Latest Version @ Cloudsmith](https://api-prd.cloudsmith.io/v1/badges/version/net-nanoframework/nanoframework-images/raw/ESP_WROVER_KIT/latest/x/?render=true)](https://cloudsmith.io/~net-nanoframework/repos/nanoframework-images/packages/detail/raw/ESP_WROVER_KIT/latest/) |
+| ESP32_PICO | | [![Latest Version @ Cloudsmith](https://api-prd.cloudsmith.io/v1/badges/version/net-nanoframework/nanoframework-images/raw/ESP32_PICO/latest/x/?render=true)](https://cloudsmith.io/~net-nanoframework/repos/nanoframework-images/packages/detail/raw/ESP32_PICO/latest/) |
+
+### ESP32 with inbuilt Ethernet
```
[the remark this answers is not in the document]
Comment (AdrianSoundy) [reply README.md:65]: As above

File: README.md (hunk 3 above)
Comment (AdrianSoundy) [reply README.md:74]: As above

File: README.md (hunk 5 above)
Comment (AdrianSoundy) [reply README.md:80]: As above

File: README.md [hunk 11]
```diff
@@ -23,43 +23,85 @@ Firmware versions contained below are released via the `main` branch, and are tr
...
+| XIAO_ESP32C3 | USB jtag -> VS| [![Latest Version @ Cloudsmith](https://api-prd.cloudsmith.io/v1/badges/version/net-nanoframework/nanoframework-images/raw/XIAO_ESP32C3/latest/x/?render=true)](https://cloudsmith.io/~net-nanoframework/repos/nanoframework-images/packages/detail/raw/XIAO_ESP32C3/latest/) |
+| ESP32_C6_Thread | USB jtag -> VS, OpenThread | [![Latest Version @ Cloudsmith](https://api-prd.cloudsmith.io/v1/badges/version/net-nanoframework/nanoframework-images/raw/ESP32_C6_THREAD/latest/x/?render=true)](https://cloudsmith.io/~net-nanoframework/repos/nanoframework-images/packages/detail/raw/ESP32_C6_THREAD/latest/) |
+| ESP32_H2_Thread | USB jtag -> VS, OpenThread | [![Latest Version @ Cloudsmith](https://api-prd.cloudsmith.io/v1/badges/version/net-nanoframework/nanoframework-images/raw/ESP32_H2_THREAD/latest/x/?render=true)](https://cloudsmith.io/~net-nanoframework/repos/nanoframework-images/packages/detail/raw/ESP32_H2_THREAD/latest/) | Esp32_H2 with openThread support
```
[the remark this answers is not in the document]
Comment (AdrianSoundy) [reply README.md:94]: Whole text should have been removed, done

File: README.md [hunk 12]
```diff
@@ -23,43 +23,85 @@ Firmware versions contained below are released via the `main` branch, and are tr
...
+
+### ESP32_S3 boards
+| Target | Note | Version | 
+|:-|---|---|
+| ESP32_S3 | Quad spiram support | [![Latest Version @ Cloudsmith](https://api-prd.cloudsmith.io/v1/badges/version/net-nanoframework/nanoframework-images/raw/ESP32_S3/latest/x/?render=true)](https://cloudsmith.io/~net-nanoframework/repos/nanoframework-images/packages/detail/raw/ESP32_S3/latest/) |
+| ESP32_S3_BLE | Display, BLE, Quad spiram support | [![Latest Version @ Cloudsmith](https://api-prd.cloudsmith.io/v1/badges/version/net-nanoframework/nanoframework-images/raw/ESP32_S3_BLE/latest/x/?render=true)](https://cloudsmith.io/~net-nanoframework/repos/nanoframework-images/packages/detail/raw/ESP32_S3_BLE/latest/) |
+| ESP32_S3_ALL |  Display, BLE, Octal spiram support | [![Latest Version @ Cloudsmith](https://api-prd.cloudsmith.io/v1/badges/version/net-nanoframework/nanoframework-images/raw/ESP32_S3_ALL/latest/x/?render=true)](https://cloudsmith.io/~net-nanoframework/repos/nanoframework-images/packages/detail/raw/ESP32_S3_ALL/latest/) |
+
+### ESP32 risc-v boards
```
[the remark this answers is not in the document]
Comment (AdrianSoundy) [reply README.md:87]: As above

**Comment by AdrianSoundy** [comment]
Readme merged and all jump links to sections working.

**Pull request merged**